tests/macho/test_builder.py

```diff
@@ -164,6 +168,24 @@ def test_add_section_id(tmp_path):
         section = lief.MachO.Section(f"__lief_{i}", [0x90] * 0x100)
         original.add_section(section)
 
+    # Check that sections are continuous and non-overlapping in each segment
+    bad_section_types = [
+            lief.MachO.Section.TYPE.ZEROFILL,
+            lief.MachO.Section.TYPE.THREAD_LOCAL_ZEROFILL,
+            lief.MachO.Section.TYPE.GB_ZEROFILL,
+            ]
+    for segment in original.segments:
+        if len(segment.sections) == 0:
+            continue
+        sections = sorted([s for s in segment.sections if s.type not in bad_section_types], key=lambda s: s.offset)
+        next_expected_offset = sections[0].offset
+        print(segment)
+        for section in sections:
+            print(section)
+            next_expected_offset = align_to(next_expected_offset, 1 << section.alignment)
+            assert section.offset == next_expected_offset
+            next_expected_offset += section.size
+
```

Comment:
I think I could move this code into `lief.MachO.check_layout`, but I am afraid of breaking lots of other tests/assumptions; this could be a follow-up.

Reply:
I currently working on updating the `check_layout` function to include latest checks from Apple. I'll add this check as well